File: OtakuNest.Client/src/app/features/comments/models/comment.dto.ts
import { ReplyDto } from "./reply.dto";

export interface CommentDto {
    id: string;
    productId: string;
    userId: string;
    userName?: string;
    content: string;
    createdAt: string;
    updatedAt?: string;
    parentCommentId?: string;
    likesCount: number;
    replies: ReplyDto[];
}

File: OtakuNest.CommentService/DTOs/CommentDto.cs
namespace OtakuNest.CommentService.DTOs
{
    public class CommentDto
    {
        public Guid Id { get; set; }
        public Guid ProductId { get; set; }
        public Guid UserId { get; set; }
        public string UserName { get; set; } = null!;
        public string Content { get; set; } = null!;
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public Guid? ParentCommentId { get; set; }
        public int LikesCount { get; set; }
        public List<ReplyDto> Replies { get; set; } = new();
    }
}
